recipes/jupyter-book/meta.yaml

```diff
@@ -0,0 +1,59 @@
+{% set name = "jupyter-book" %}
+{% set version = "0.7.5" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 92bb13b517568498e5176c62d088ec9430a9059ee6505414ca7663405ceb1686
+
+build:
+  number: 0
+  entry_points:
+    - jb = jupyter_book.commands:main
+    - jupyter-book = jupyter_book.commands:main
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - attrs >=19.3,<20
+    - click
+    - docutils >=0.15
+    - myst-nb >=0.8.0,<0.9
+    - nbconvert
+    - nbformat
+    - python >=3.6
+    - pyyaml
+    - sphinx <3
+    - sphinx-comments
+    - sphinx-copybutton
+    - sphinx-thebe >=0.0.6
+    - sphinx_book_theme >=0.0.34
+    - sphinx_togglebutton
+    - sphinxcontrib-bibtex
+
```

Comment:
It might also be nice to add [outputs](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#outputs-section) to support capturing the pdf output target... e.g. `jupyter-book-pdf`...

Reply:
Can you explain this further thanks, I'm not quite sure what added benefit this brings?

Reply:
Conda doesn't support the `[extras]` section that pip does (which ends up being kinda weird, anyway, as it isn't respected by `pip check`, nor will extra pins be taking into account for later solves, but anyway...) so if folk want the pdf output option (which is great) they'll get the nastygram about installing `pypetteer`, etc. If a lot of people want pdf, this might be a sensible default, but that toolchain has problems of its own (downloading chromium at runtime), so an easy out would be useful

concretely, there could be a `jupyter-book-core`, which contains all the hard dependencies, and a `jupyter-book-webpdf` which depends on `-core` and `pypetteer` (with the appropriate version), while `jupyter-book` could depend on `-core` and `-pdf`.

```yaml
outputs:
- name: jupyter-book
  requirements:
    run: 
      - jupyter-book-core ={{ version }}
      - jupyter-book-webpdf ={{ version }}
- name: jupyter-book-core
  requirements:
    run:  # all the reqs
 - name: jupyter-book-webpdf
  requirements:
    run: 
      - jupyter-book-core ={{ version }}
      - pypeteer
```

Of course, this would be better if there was an `nbformat-webpdf` we could depend on... but the [`run_constrained`](https://github.com/conda-forge/nbconvert-feedstock/blob/master/recipe/meta.yaml#L38) will at least take care of the version, so the pin wouldn't have to track here.

Reply:
Oh yeh ta I forgot that was under `outputs`. I do something similar on another project https://github.com/conda-forge/aiida-core-feedstock/blob/master/recipe/meta.yaml

Reply:
yeah, outputs are kinda gross, but the best we have today...

Reply:
This looks good to me, and would add some nice flexibility.